packages/marketplace/src/components/pages/apps/apps.tsx

```diff
@@ -32,64 +36,65 @@ export const handleOnCardClick = (history: History) => (app: AppSummaryModel) =>
   history.push(`${Routes.APPS}/${app.id}`)
 }
 
+export const handleLoadMore = (dispatch: Dispatch) => (page: number) => {
+  console.log('[handleLoadMore]', page)
+  dispatch(clientFetchAppSummary({ page }))
+}
+
 export const Apps: React.FunctionComponent = () => {
   const history = useHistory()
   const location = useLocation()
+  const dispatch = useDispatch()
 
   const appSummaryState = useSelector(selectAppSummary)
-  const urlParams = new URLSearchParams(location.search)
-  const page = Number(urlParams.get('page')) || 1
-
   const hasParams = hasFilterParams(location.search)
-  const unfetched = !appSummaryState.data
   const loading = appSummaryState.isAppSummaryLoading
   const apps = appSummaryState?.data?.apps?.data || []
   const featuredApps = appSummaryState?.data?.featuredApps || []
-  const { totalCount, pageSize } = appSummaryState?.data?.apps || {}
+  const { totalCount, pageNumber = 1 } = appSummaryState?.data?.apps || {}
+
+  const hasMore = !apps || apps.length == 0 || loading ? false : pageNumber * 9 < (totalCount || 0)
```

Comment:
Feels like this logic could be moved into it's own function and be tested

Reply:
Just added some comments for this statement